### ISSUE TITLE
Login page connecting to Cognito

### ISSUE THREAD

**Issue by coultergeist** [ActoKids/AD440_W19_CloudPracticum#65]
Connect the login page to Cognito (working with Zach from API team).

Once assigned, annotate your estimated time. Then, keep regular updates on your issue. Comment your approach, tutorials, articles, updates, links, photos, etc. Comment often. When closing your issue include the following in your final comment.

Time estimated: 3 hours
Time spent:
Pull Request link:
Wiki link:
Deployed Code link: 
Testers:
What I tested:

**Comment by coultergeist** [comment]
Issue #77 created 26Feb2019 for AWS permissions (including Cognito)

Tutorials to start:
- https://www.youtube.com/watch?v=-ZrYlsEBLmo
- https://github.com/mavi888/web-client-cognito

May try this: 
```
// Initialize the Amazon Cognito credentials provider
CognitoCachingCredentialsProvider credentialsProvider = new CognitoCachingCredentialsProvider(
   getApplicationContext(),
   "us-east-1:163c7994-ce5d-480c-bd4a-86045414024d", // Identity pool ID
   Regions.US_EAST_1 // Region
);
```

**Comment by coultergeist** [comment]
Unable to create user pools in Cognito (see bug #77). 

AWS>Cognito>Create User Pool>name pool>Review defaults....>Create then I see this error

<img width="953" alt="cognito-error" src="https://user-images.githubusercontent.com/16545169/53687870-1e9d8200-3cef-11e9-9863-4b505df3526f.png">

**Comment by coultergeist** [comment]
Hours to date: 7

Still incorrect Cognito perms. Continuing to work with DevOps (see bug #77)
<img width="1212" alt="screen shot 2019-03-05 at 2 13 40 pm" src="https://user-images.githubusercontent.com/16545169/53841341-69b6cf80-3f51-11e9-9483-c4b95ee84364.png">

There is a configured version on the ActoKids AWS account under API Team's Cognito (@ZakBrinlee's AWS Congito) See [bug #87 ](https://github.com/ActoKids/AD440_W19_CloudPracticum/issues/87)

**Comment by coultergeist** [comment]
Hours to date: 10

Cognito perms 50% complete:

There are two Cognito functions:

1) "Manage User Pools" - I am able to fully create a user pool.
2) "Manage Identity Pools" - Not authorized
`User: arn:aws:iam::061431082068:user/danielle.coulter@seattlecolleges.edu is not authorized to perform: iam:CreateRole on resource: arn:aws:iam::061431082068:role/Cognito_ActoKidsAuth_Role`

Note: I (believe) I need DevOps to create a policy and a role that allows the people logging in to actually invoke the API.

**Comment by coultergeist** [comment]
Hours to Date: 11

### Coding for User Login
Complete. Right now users can successfully log in (with API Team Cognito User Pool)

exerpt from `config.js` file:
`/* This is the UI Team Cognito Info. Remove above cognito info an uncomment this out`

`cognito: {`
		`REGION: 'us-east-2',`
		`USER_POOL_ID: 'us-east-2_otpHbMxKi',`
		`APP_CLIENT_ID: '7l7q1utlnmg3qnmlffc5n4otin',`
		`IDENTITY_POOL_ID: 'us-east-2:dee596e1-548a-466c-be33-e9bc5951811f'
	`} */`

<img width="1038" alt="login" src="https://user-images.githubusercontent.com/16545169/53915015-ddb9ac00-4013-11e9-89a8-0d232ff65122.png">

Once logged in, the user is successfully directed to Dashboard
<img width="1023" alt="dashboard" src="https://user-images.githubusercontent.com/16545169/53915030-ea3e0480-4013-11e9-8243-f0fb84dacc63.png">

**Comment by coultergeist** [comment]
Time estimated: 3 hours
Time spent: 16 hours 
Pull Request link:

- [PR 30: Add cognito user pool connection info](https://github.com/ActoKids/web-ui/pull/30)
- [PR 31: Fix Signin](https://github.com/ActoKids/web-ui/pull/31)

Wiki link: [Configuring Cognito Page for User Login](https://github.com/ActoKids/AD440_W19_CloudPracticum/wiki/Configuring-Cognito-for-User-Login)
Deployed Code link: [2edusite](http://www.2edusite.com)

Testers:

- API Team. See [Bug 87: Confirm Cognito config UITeam/API](https://github.com/ActoKids/AD440_W19_CloudPracticum/issues/87) for feedback by @ZakBrinlee 

What I tested:

- Pull Request for [Erik Feature #25:](https://github.com/ActoKids/web-ui/pull/25) pinned to [Issue #47 UI Forms:dashboard>GET/events](https://github.com/ActoKids/web-ui/pull/25)
- Bug: Closed [Issue #77](https://github.com/ActoKids/AD440_W19_CloudPracticum/issues/77):(AWS perms itemization and request) with DevOps @LyndonP
- Bug: Open [Issue #26](https://github.com/ActoKids/web-ui/issues/26): Identify filtering criteria with API Team @MrSamsa42
- Bug: Closed [Issue 87 Confirm Cognito config between UITeam/API Team](https://github.com/ActoKids/AD440_W19_CloudPracticum/issues/87)
- Pull Request for [Erik Feature #27](https://github.com/ActoKids/web-ui/pull/27) (reviewer)

Additional Notes: 
- Currently, there is one "user"
- Our next steps are to implement multi-users (Org and Super Admin) which will require that we revisit Cognito again.
